src/CosmosLogStorage.test.ts

```diff
@@ -0,0 +1,211 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import * as CLM from '@conversationlearner/models';
+import { CosmosLogStorage } from './CosmosLogStorage';
+
+const options = {
+    endpoint: "",
+    key: ""
+}
+
+function makeLogDialog(logDialogId: string): CLM.LogDialog {
+    return {
+        ...CLM.MockData.makeLogDialog(),
+        logDialogId
+    }
+}
+
+async function getStorage(): Promise<CosmosLogStorage | undefined> {
+    if (!options.endpoint || !options.key) {
+        console.log("Skipping Test.  Cosmos credentials not defined.")
+        return undefined
+    }
+    return await CosmosLogStorage.Get(options)
+}
+
+async function appendScorerStep(appId: string, logDialogId: string, logScorerStep: CLM.LogScorerStep, cls: CosmosLogStorage) {
+    const logDialog: CLM.LogDialog | undefined = await cls.Get(appId, logDialogId)
+    if (!logDialog) {
+        throw new Error(`Log Dialog does not exist App:${appId} Log:${logDialogId}`)
+    }
+    const lastRound = logDialog.rounds[logDialog.rounds.length - 1]
+    if (!lastRound || !lastRound.extractorStep) {
+        throw new Error(`Log Dialogs has no Extractor Step Id:${logDialogId}`)
+    }
+    lastRound.scorerSteps.push(logScorerStep as any)
+    logDialog.dialogEndDatetime = new Date().toJSON()
+    await cls.Replace(appId, logDialog)
+}
+
+async function appendExtractorStep(appId: string, logDialogId: string, extractorStep: CLM.LogExtractorStep, cls: CosmosLogStorage) {
+    // Append an extractor step to already existing log dialog
+    const logDialog: CLM.LogDialog | undefined = await cls.Get(appId, logDialogId)
+    if (!logDialog) {
+        throw new Error(`Log Dialog does not exist App:${appId} Id:${logDialogId}`)
+    }
+    const newRound: CLM.LogRound = {
+        extractorStep: { ...extractorStep, stepBeginDatetime: "", stepEndDatetime: "" },
+        scorerSteps: []
+    }
+    logDialog.rounds.push(newRound)
+    logDialog.dialogEndDatetime = new Date().toJSON()
+    await cls.Replace(appId, logDialog)
+}
+
+describe('CosmosLogStorage', () => {
```

Comment:
>describe [](start = 0, length = 8)

is there a decorator or method that make these tests manual? I guess current structure causes these tests to be discovered and run automatically by unittest framework and since there is not key or endpoint, these will fail.

Reply:
Anything names *.test.ts is run automatically.  There's some code in here that skips these tests (with a warning) if the endpoint/key isn't defined